metadata/metadata.go

```diff
@@ -156,28 +188,34 @@ var bannedStrings = [...]string{"__"}
 var bannedPrefixes = [...]string{"_"}
 var bannedSuffixes = [...]string{"_"}
 
-func resourceNamedSafely(id ResourceID) error {
+func resourceNamedSafely(id ResourceID, logger logging.Logger) error {
```

Comment:
Actually, lets not pass in the logger here. Feels like the caller of this should just handle the error and log